### DISCUSSION TITLE
CancellationToken in DocumentFormat.OpemXml.Packaging.SpreadsheetDocument.Open function

### DISCUSSION THREAD

**Discussion by ahsanaman92** [OfficeDev/Open-XML-SDK#1489] [General]
Version: DocumentFormat.OpenXml v2.20.0

I am working on an issue where the customer has reported that it takes a while for the _DocumentFormat.OpemXml.Packaging.SpreadsheetDocument.Open_ function to determine a large Excel document (329 MB) is invalid over the network. Is there a plan to be able to pass a CancellationToken to the open function? 

The function is passed a _System.IO.Stream input_ like this: using (SpreadsheetDocument Doc = SpreadsheetDocument.Open(input, false))

**Comment by AlfredHellstern** [comment]
opened issue [#1490 ]([[url](https://github.com/dotnet/Open-XML-SDK/issues/1490)])for this
